src/migrate.rs

```diff
@@ -0,0 +1,139 @@
+// Config.json migration module
+//
+// Provides methods for migrating config.json fields based on remote directives
+// from /os/vX/config. Limits migrated fields based on os-config.json schema
+// whitelist.
+
+use crate::config_json::ConfigMap;
+use crate::remote::ConfigMigrationInstructions;
+use crate::schema::OsConfigSchema;
+use anyhow::Result;
+
+pub fn generate_config_json_migration(
+    schema: &OsConfigSchema,
+    migration_config: &ConfigMigrationInstructions,
+    config_json: &ConfigMap,
+) -> Result<ConfigMap> {
+    info!("Checking for config.json migrations...");
+
+    let mut new_config = config_json
+        .iter()
+        .map(|(k, v)| (k.clone(), v.clone()))
+        .collect::<ConfigMap>();
```

Comment:
Is not that equivalent of simply `let mut new_config = config_json.clone();`?

Reply:
You're right 😅